### DISCUSSION TITLE
Moosefs Replication between two DC

### DISCUSSION THREAD

**Discussion by kiranranjane** [moosefs/moosefs#586] [General]
Is it possible to replicated data between two datacenter using moosefs, For example one DC is in New York and another is in Chicago then how could we sync the data between these DC and keep all metadata and chunks in sync between them.

Does Moosefs have feature of asynchronous replication between sites?

Thanks

**Comment by borkd** [comment]
Short answer: in many cases, yes.

You could configure clients in each DC to write to redundant storage classes served by a set of local chunkservers, and switch to storage class replicated across multiple DCs after a period of time. 

Network between DCs should be fast enough with respect of the volume of new/changed data to allow the replication to catch up, even in the face of maintenance or failures which could result inter-DC traffic. This requirement is not specific to MooseFS though.

Start with replication levels which let you do maintenance or tolerate failures within a DC without needing to pull data from other DCs.

Then do some homework w.r.t current and planned workloads:
- Size distribution of modified files. Replication of bazillions of small files will be latency bound and has to be tuned for that edge case.
- Frequency of changes and volume of changed data per unit of time. When latency is accounted for how long will it take to replicate changes? If on paper your are barely able to catch up, there is a good chance something important will be missing when you need it the most (sudden and extended outage in one of the DCs causing abnormal traffic patterns).

Hope this helps
